extensions/resource-deployment/src/services/notebookService.ts

```diff
@@ -25,18 +24,22 @@ export class NotebookService implements INotebookService {
 	 * Copy the notebook to the user's home directory and launch the notebook from there.
 	 * @param notebook the path of the notebook
 	 */
-	launchNotebook(notebook: string | NotebookInfo): void {
+	launchNotebook(notebook: string | NotebookInfo): Thenable<azdata.nb.NotebookEditor> {
 		const notebookPath = this.getNotebook(notebook);
 		const notebookFullPath = path.join(this.extensionPath, notebookPath);
-		if (notebookPath && this.platformService.fileExists(notebookPath)) {
-			this.showNotebookAsUntitled(notebookPath);
-		}
-		else if (notebookPath && this.platformService.fileExists(notebookFullPath)) {
-			this.showNotebookAsUntitled(notebookFullPath);
-		}
-		else {
-			this.platformService.showErrorMessage(localize('resourceDeployment.notebookNotFound', "The notebook {0} does not exist", notebookPath));
-		}
+		return this.platformService.fileExists(notebookPath).then((notebookPathExists) => {
+			if (notebookPathExists) {
+				return this.showNotebookAsUntitled(notebookPath);
+			} else {
+				return this.platformService.fileExists(notebookFullPath).then(notebookFullPathExists => {
+					if (notebookFullPathExists) {
+						return this.showNotebookAsUntitled(notebookFullPath);
+					} else {
+						throw localize('resourceDeployment.notebookNotFound', "The notebook {0} does not exist", notebookPath);
+					}
```

Comment:
Typically we haven't been localizing errors since we don't expect those to be shown to the user. Instead we'd want to catch it at some level and display a message specifically there

Reply:
potentially this error could be shown to the user, we have a command to allow user open a dialog, collect input and open notebook. azdata.openNotebookInputDialog

Reply:
Not sure what you mean. But given that 99.9% of our uses of throw is with non-localized values I don't see why this one should be different. As I said if you want to display a message to the user in a localized format then it's best to catch errors and display the message at that time usually.